client/src/plus/collections-quicksearch.ts

```diff
@@ -0,0 +1,73 @@
+const COLLECTION_ITEMS_KEY: string = "collection-items";
+const COLLECTION_ITEMS_UPDATED_DATE_KEY: string =
+  "collection-items-updated-date";
+
+type CollectionItem = {
+  id: number;
+  title: string;
+  url: string;
+};
+
+function getCollectionItemsUpdatedDate(): Date | null {
+  let dateString = window?.localStorage?.getItem(
+    COLLECTION_ITEMS_UPDATED_DATE_KEY
+  );
+  if (!dateString) {
+    return null;
+  }
+  return new Date(dateString);
+}
+
+function setCollectionItemsUpdatedDate(updated: Date) {
+  window?.localStorage?.setItem(
+    COLLECTION_ITEMS_UPDATED_DATE_KEY,
+    updated.toISOString()
+  );
+}
+
+export function getCollectionItems(): CollectionItem[] {
+  return JSON.parse(
+    window?.localStorage?.getItem(COLLECTION_ITEMS_KEY) || "[]"
+  );
+}
+
+function setCollectionItems(items: CollectionItem[]) {
+  window?.localStorage?.setItem(COLLECTION_ITEMS_KEY, JSON.stringify(items));
+  window?.mdnWorker?.mutationCounter ?? window.mdnWorker.mutationCounter++;
+}
+
+export async function fetchAllCollectionsItems(remoteUpdated: null | Date) {
+  const localUpdated = getCollectionItemsUpdatedDate();
+  if (!localUpdated || !remoteUpdated || remoteUpdated > localUpdated) {
+    const res = await fetch("/api/v1/plus/collection/?limit=999");
+    const { items = [] } = await res.json();
+    setCollectionItems(items);
+    setCollectionItemsUpdatedDate(remoteUpdated || new Date());
+  }
+}
+
+export function addCollectionItem({ id, title, url }: CollectionItem) {
```

Comment:
This function is still not used anywhere, will it be?

Reply:
They'll be used in the collection-v2 branch.